apps/frontend/src/pages/construct-route/construct-route.tsx

```diff
@@ -0,0 +1,65 @@
+import { MapProvider } from "~/libs/components/components.js";
+import {
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useMemo,
+	useState,
+} from "~/libs/hooks/hooks.js";
+import { type RouteLine } from "~/libs/types/types.js";
+import { type PointsOfInterestGetAllItemResponseDto } from "~/modules/points-of-interest/points-of-interest.js";
+
+import { SidePanel } from "./libs/components/side-panel/side-panel.js";
+import styles from "./styles.module.css";
+
+const ConstructRoute = (): React.JSX.Element => {
+	const { routeLineString } = useAppSelector((state) => state.constructRoute);
```

Comment:
It's better to access properties the way we store/update them for useAppSelector. Without additional memoization helpers, useAppSelector will trigger component re-render whenever changes in the constructRoute object appear, while we want the re-render only if routeLineString has changed, not any other property in the store inside constructRoute
```suggestion
	const routeLineString = useAppSelector((state) => state.constructRoute.routeLineString);
```